gapii/cc/call_observer.h

```diff
@@ -248,7 +248,11 @@ inline Slice<T> CallObserver::copy(const Slice<T>& dst, const Slice<T>& src) {
 template <typename T>
 inline Slice<T> CallObserver::clone(const Slice<T>& src) {
     Slice<T> dst = make<T>(src.count());
-    copy(dst, src);
+    // Make sure that we actually fill the data the first time.
+    // If we use ::copy(), then the copy will only happen if
+    // the observer is active.
```

Comment:
So there are a couple of places where this is needed. I will provide an example with VkShaderModule/ShaderModuleObject.

How we used to handle VkCreateShaderModule.

```
//*.api_types.tmpl
{{define "DeclareClass"}}
{{$is_shader_module := (eq $.Name "ShaderModuleObject")}}
{{if $is_shader_module}}
   std::vector<uint32_t> shaderWords;
{{end}}
// C++ Extra
void VulkanSpy::addWords(CallObserver* observer, VkShaderModule module, size_val count, uint32_t* words) {
   auto& m = ShaderModules[module];
   m->shaderWords = std::vector<uint32_t>(words, words+count/4);
}
``` 
So basically, we would create a hidden std::vector on the side of the state object. However this is not "officially" in the state block. This was done because clone() does not actually do the copy when we are not tracking memory operations, so we would end up dropping the words on the floor for MEC. We ended up stuffing this in a side data-structure so that we could guarantee the write.

If we want to remove it, we need that clone to actually copy the bytes, even when observations are turned off. However turning on ALL copies drastically slows down the performance during MEC. (To the point that in some scenes the program is almost unusable), so the compromise was the assumption that a clone is important, and keeps data even when tracing is off, but a copy can be elided.

Another option might be a way to annotate a slice with "Really actually do the copy".